composable_kernel/include/tensor_operation/element_wise_operation.hpp

```diff
@@ -141,6 +141,93 @@ struct AddHardswishAdd
     }
 };
 
+// Unary operators are usually called element-wisely before the reduction is executed on the
+// elements.
+// They are needed for easy implementation of reduction types of AVG, NRM1, NRM2
+
+template <typename Y, typename X, bool HasDividing = false>
+struct UnaryIdentic
+{
+    __host__ __device__ UnaryIdentic(const index_t divider = 1) { (void)divider; };
+
+    __host__ __device__ constexpr void operator()(Y& y, const X& x) const
+    {
+        y = type_convert<Y>(x);
```

Comment:
We want to avoid unwanted type casting.

template <typename Y, typename X>
struct PointwiseOp;

template<>
struct PointwiseOp<float, half_t>
{
   __host__ __device__ constexpr void operator()(float& y, const half_t& x) const
    {
        // do something;
    }
};

Reply:
`half_t` to `float` does need explicit type converting.  Do you mean provide  instantiation for `<float, float>`, `<half_t, half_t>` ?

Reply:
Yes, this is to avoid any type conversion when calling the function